mlflow/projects/pollable_run.py

```diff
@@ -0,0 +1,149 @@
+from abc import abstractmethod
+import os
+import signal
+import subprocess
+import sys
+
+from mlflow.entities.run_status import RunStatus
+from mlflow.utils.logging_utils import eprint
+
+
+def maybe_set_run_terminated(active_run, status):
+    """
+    If the passed-in active run is defined and still running (i.e. hasn't already been terminated
+    within user code), mark it as terminated with the passed-in status.
+    """
+    if active_run and not RunStatus.is_terminated(active_run.get_run().info.status):
+        active_run.set_terminated(status)
+
+
+def monitor_run(pollable_run, active_run):
```

Comment:
Should this be defined in `submitted_run`? It's only used there, and may make it easier to understand the goal of `submitted_run` (which is twofold, to provide an API to users to interact with the run, and to fork the monitoring subprocess to do this logic).